burn-import/src/burn/node/base.rs

```diff
@@ -178,17 +178,60 @@ impl<PS: PrecisionSettings> NodeCodegen<PS> for Node<PS> {
 }
 
 #[cfg(test)]
-mod tests {
+pub(crate) mod tests {
     use crate::burn::{
         graph::BurnGraph,
-        node::{conv2d::Conv2dNode, matmul::MatmulNode, test::assert_tokens},
+        node::{conv2d::Conv2dNode, matmul::MatmulNode, test::assert_tokens, NodeCodegen},
         TensorType,
     };
     use burn::{
         nn::conv::Conv2dConfig, nn::PaddingConfig2d, record::FullPrecisionSettings, tensor::Data,
     };
+    use proc_macro2::TokenStream;
     use quote::quote;
 
+    fn one_node_graph<T: NodeCodegen<FullPrecisionSettings> + 'static>(
+        node_gen: T,
+    ) -> BurnGraph<FullPrecisionSettings> {
+        let mut graph = BurnGraph::<FullPrecisionSettings>::default();
+
+        graph.register(node_gen);
+
+        graph
+    }
+
+    fn unary_operator_expected(function: TokenStream) -> TokenStream {
+        quote! {
+            use burn::{
+                module::Module,
+                tensor::{backend::Backend, Tensor},
+            };
+
+            #[derive(Module, Debug)]
+            pub struct Model <B: Backend>{}
+
+            impl<B: Backend> Model <B> {
+                pub fn new_with(record: ModelRecord<B>) -> Self {
+                    Self { }
+                }
+                #[allow(clippy::let_and_return)]
+                pub fn forward(&self, tensor1: Tensor<B, 4>) -> Tensor<B, 4> {
```

Comment:
I would use a const generic instead of 4 here. `unary_operator_expected<const D: usize>`

Reply:
Unfortunately if we use const generics, tests do not pass:

```
---- burn::node::sigmoid::tests::test_codegen_node stdout ----
thread 'burn::node::sigmoid::tests::test_codegen_node' panicked at 'assertion failed: `(left == right)`

Diff < left / right > :
 use burn::{
     module::Module,
     tensor::{backend::Backend, Tensor},
 };
 #[derive(Module, Debug)]
 pub struct Model<B: Backend> {}
 impl<B: Backend> Model<B> {
     pub fn new_with(record: ModelRecord<B>) -> Self {
         Self {}
     }
     #[allow(clippy::let_and_return)]
<    pub fn forward(&self, tensor1: Tensor<B, 4>) -> Tensor<B, 4> {
>    pub fn forward(&self, tensor1: Tensor<B, 4usize>) -> Tensor<B, 4usize> {
         let tensor2 = tensor1.sigmoid();
         tensor2
     }
 }
```